python/tk_multi_loader/loader_action_manager.py

```diff
@@ -105,40 +81,133 @@ def get_actions_for_publish(self, sg_data, ui_area):
         else:
             raise TankError("Unsupported UI_AREA. Contact support.")
 
-        # convert created_at unix time stamp to shotgun std time stamp
+        # convert created_at unix time stamp to shotgun time stamp
         unix_timestamp = sg_data.get("created_at")
-        if unix_timestamp:
+        if isinstance(unix_timestamp, float):
             sg_timestamp = datetime.datetime.fromtimestamp(unix_timestamp, 
                                                            shotgun_api3.sg_timezone.LocalTimezone())
             sg_data["created_at"] = sg_timestamp
-                    
-        action_defs = []
+
+        action_defs_dict = {}
         try:
-            action_defs = self._app.execute_hook_method("actions_hook", 
-                                                        "generate_actions", 
-                                                        sg_publish_data=sg_data, 
-                                                        actions=actions,
-                                                        ui_area=ui_area_str)
+            # call out to hook to give us the specifics.
+            action_defs_list = self._app.execute_hook_method("actions_hook",
+                                                       "generate_actions",
+                                                       sg_publish_data=sg_data,
+                                                       actions=actions,
+                                                       ui_area=ui_area_str)
+            for action_def in action_defs_list:
+                action_defs_dict[action_def["name"]] = action_def
+
         except Exception:
             self._app.log_exception("Could not execute generate_actions hook.")
-            
-            
-            
-        # create QActions
+
+        return action_defs_dict
+
+    def get_actions_for_publishes(self, sg_data_list, ui_area):
```

Comment:
This is the new entry point on the action manager to get actions for a given selection.